### PR TITLE
Replace result enum with single struct

### PR DESCRIPTION
**Pull request by maciektr**
**Stack**:
- #1260
- #1249 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Comment by mkaput** [comment]
Please rebase

File: scarb/tests/build_cairo_plugin.rs [hunk 1]
```diff
@@ -370,7 +369,7 @@ fn can_remove_original_node() {
 
         #[attribute_macro]
         pub fn some(_: TokenStream) -> ProcMacroResult {
-            ProcMacroResult::Remove { diagnostics: Vec::new() }
+            ProcMacroResult::new(TokenStream::new("".to_string()))
```
Comment (mkaput) [inline scarb/tests/build_cairo_plugin.rs:372]:
```suggestion
            ProcMacroResult::new(TokenStream::empty()))
```

wdyt?

File: scarb/tests/build_cairo_plugin.rs
Comment (mkaput) [inline scarb/tests/build_cairo_plugin.rs:1]: Please add a test that attempts to remove inline macro

**Review by mkaput — CHANGES_REQUESTED** [review]

**Comment by maciektr** [comment]
Closing pull request: commit has gone away

**Pull request closed without merging**